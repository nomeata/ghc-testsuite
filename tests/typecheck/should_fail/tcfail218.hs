{-# LANGUAGE IncoherentInstances, MultiParamTypeClasses, FlexibleInstances #-}

import Tcfail218_Help

instance C [a] b where foo = undefined
instance C a Int where foo = undefined

-- Should fail, as a more specific, unifying but not matching, non-incoherent instance exists.
x :: ([a],b)
x = foo

main = return ()
