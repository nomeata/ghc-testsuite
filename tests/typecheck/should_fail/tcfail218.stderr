
tcfail218.hs:10:5:
    Overlapping instances for C [a] b arising from a use of ‛foo’
    Matching instances:
      instance [incoherent] C [a] b -- Defined at tcfail218.hs:5:10
      instance C [Int] b -- Defined at Tcfail218_Help.hs:7:10
    (The choice depends on the instantiation of ‛a, b’
     To pick the first instance above, use IncoherentInstances
     when compiling the other instance declarations)
    In the expression: foo
    In an equation for ‛x’: x = foo
